syntax = "proto3";
package teacher_out;

service TeacherOut {
  rpc CreateTeacher(CreateTeacherRequest) returns (CreateTeacherResponse) {}
  rpc ReadTeacher(ReadTeacherRequest) returns (ReadTeacherResponse) {}
  rpc ReadTeacherList(ReadListTeacherRequest) returns (ReadListTeacherResponse) {}
  rpc UpdateTeacher(UpdateTeacherRequest) returns (UpdateTeacherResponse) {}
  rpc DeleteTeacher(DeleteTeacherRequest) returns (DeleteTeacherResponse) {}
}

message CreateTeacherRequest {
  string first_name = 1;
  string last_name = 2;
  string phone_number = 3;
  string email = 4;
}

message CreateTeacherResponse { string msg = 1; }

message ReadTeacherRequest { int32 id = 1; }

message ReadTeacherResponse {
  string first_name = 1;
  string last_name = 2;
  string phone_number = 3;
  string email = 4;
}

message UpdateTeacherRequest {
  int32 id = 1;
  string first_name = 2;
  string last_name = 3;
  string phone_number = 4;
  string email = 5;
}

message UpdateTeacherResponse { string msg = 1; }

message DeleteTeacherRequest { int32 id = 1; }

message DeleteTeacherResponse { string msg = 1; }

message ReadListTeacherRequest {}

message ReadListTeacherResponse { repeated ReadTeacherResponse teachers = 1; }